q2_annotate/busco/busco.py

```diff
@@ -21,93 +20,71 @@
+    # mags.
+    base_cmd = ["busco", "-f", *params]
+
+    cmd = deepcopy(base_cmd)
+    cmd.extend([
+        "--in",
+        input_dir,
+        "--out_path",
+        output_dir,
+        "-o",
+        sample
+    ])
+    run_command(cmd,  cwd=os.path.dirname(output_dir))
 
-    if isinstance(mags, MultiMAGSequencesDirFmt):
-        manifest: pd.DataFrame = mags.manifest.view(pd.DataFrame)
-        manifest["sample_dir"] = manifest.filename.apply(
-            lambda x: os.path.dirname(x)
-        )
-        sample_dirs = manifest["sample_dir"].unique()
 
-    elif isinstance(mags, MAGSequencesDirFmt):
-        sample_dirs = [str(mags)]
-
-    path_to_run_summaries = {}
-
-    # For every unique sample dir run busco
-    for sample_dir in sample_dirs:
-        sample = os.path.split(sample_dir)[-1]
-
-        cmd = deepcopy(base_cmd)
-        cmd.extend([
-            "--in",
-            sample_dir,
-            "--out_path",
-            output_dir,
-            "-o",
-            sample
-        ])
-        run_command(cmd,  cwd=os.path.dirname(output_dir))
-
-        path_to_run_summary = os.path.join(
-            output_dir, sample, "batch_summary.txt"
-        )
-        if os.path.isfile(path_to_run_summary):
-            path_to_run_summaries[sample] = path_to_run_summary
-        else:
-            raise FileNotFoundError(
-                f"BUSCO batch summary file {path_to_run_summary} not found."
-            )
+def _busco_helper(mags, common_args):
+    results_all = pd.DataFrame()
 
-    return path_to_run_summaries
+    if isinstance(mags, MultiMAGSequencesDirFmt):
+        sample_dir = mags.sample_dict()
 
+    elif isinstance(mags, MAGSequencesDirFmt):
+        sample_dir = {"": mags.feature_dict()}
 
-def _busco_helper(mags, common_args):
     with tempfile.TemporaryDirectory() as tmp:
-        path_to_run_summaries = _run_busco(
-            output_dir=os.path.join(tmp, "busco_output"),
-            mags=mags,
-            params=common_args,
-        )
+        for sample_id, feature_dict in sample_dir.items():
 
-        all_summaries = _collect_summaries(
-            run_summaries_fp_map=path_to_run_summaries,
-        )
-    all_summaries = _rename_columns(all_summaries)
+            _run_busco(
+                input_dir=os.path.join(str(mags), sample_id),
+                output_dir=str(tmp),
+                sample=sample_id,
+                params=common_args,
+            )
+            # Extract results from JSON files for one sample
+            for mag_id, mag_fp in feature_dict.items():
+                results_mag = _extract_json_data(
+                    str(tmp), mag_id, sample_id, os.path.basename(mag_fp)
+                )
 
-    lengths = _get_mag_lengths(mags)
-    all_summaries = all_summaries.join(lengths, on="mag_id")
+                # Add results df from one sample to results_all df
+                results_all = pd.concat([results_all, results_mag])
```

Comment:
You should probably only collect all the dfs into a list and concatenate them once at the very end, instead of concatenating at every iteration.